File: Application/src/main/java/com/shareyourproxy/widget/ContentDescriptionDrawable.java
package com.shareyourproxy.widget;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;

/**
 * This BitmapDrawable has a name. For now this functionality is being used to display the {@link
 * ContentDescriptionDrawable#_contentDescription} in a Toast when a user long presses a tab {@link
 * android.support.design.widget.TabLayout}.
 */
public class ContentDescriptionDrawable extends BitmapDrawable {

    private String _contentDescription = "";

    /**
     * Create image.drawable from a bitmap, setting initial target density based on the display
     * metrics of the resources.
     *
     * @param res display metrics
     * @param bmp Bitmap
     */
    public ContentDescriptionDrawable(Resources res, Bitmap bmp) {
        super(res, bmp);
    }

    /**
     * Getter.
     *
     * @return content description
     */
    public String getContentDescription() {
        return _contentDescription;
    }

    /**
     * Setter.
     *
     * @param contentDescription description of image.drawable
     * @return this
     */
    public ContentDescriptionDrawable setContentDescription(String contentDescription) {
        this._contentDescription = contentDescription;
        return this;
    }

}

File: Application/src/main/kotlin/com/shareyourproxy/widget/ContentDescriptionDrawable.kt
package com.shareyourproxy.widget

import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable

/**
 * This BitmapDrawable has a name. For now this functionality is being used to display the [ContentDescriptionDrawable.contentDescription] in a Toast
 * when a user long presses a tab [android.support.design.widget.TabLayout].
 * Create an image from a bitmap, setting initial target density based on the display metrics of the resources.
 * @param res display metrics
 * @param bmp Bitmap
 */
internal final class ContentDescriptionDrawable(res: Resources, bmp: Bitmap) : BitmapDrawable(res, bmp) {
    internal var contentDescription = ""
    internal fun setContentDescription(string: String): ContentDescriptionDrawable {
        contentDescription = string
        return this
    }
}
